src/main/java/com/codeus/winter/config/DefaultBeanFactory.java

```diff
@@ -451,6 +453,24 @@ protected Object resolveBean(DependencyDescriptor descriptor) {
         return getBean(targetCandidate.getKey(), targetCandidate.getValue());
     }
 
+    private Map.Entry<String, BeanDefinition> determinePrimaryCandidate(
+            List<Map.Entry<String, BeanDefinition>> candidates
+    ) {
+        Map.Entry<String, BeanDefinition> result = null;
+        int primaryAnnotationCount = 0;
+        for (Map.Entry<String, BeanDefinition> candidate : candidates) {
+            var beanDefinition = candidate.getValue();
+            if (beanDefinition.isPrimary()) {
+                primaryAnnotationCount++;
+                if (primaryAnnotationCount > 1) {
+                    return null;
+                }
```

Comment:
Please add one more test for this case (basically you need to have two beans with the `@Primary` annotation) and expect that `NotUniqueBeanDefinitionException` is thrown (from the `DefaultBeanFactory#resolveBean` as the result of that multiple candidates but the factory cannot resolve one).

Reply:
Added